aten/src/ATen/function_wrapper.py

```diff
@@ -1536,10 +1536,15 @@ def process_legacy_th_option(option):
             env = nested_dict(option, backend_type_env)
             body = emit_body(env, option, option['backend_types'][backend])  # type: ignore
             option['type_definition_body'] = body
+            # These type ignores arise from the fact that a nested_dict
+            # technically isn't a Mapping, as it doesn't implement
+            # enough methods.  I could fix this with a Protocol but
+            # then I need typing_extensions which isn't currently
+            # a build dep.
```

Comment:
That's the second time today `typing_extensions` would've come in handy. I opened https://github.com/pytorch/pytorch/issues/38221 for adding it a while ago, let me know if you want me to open a PR for that.